app/liquid_tags/classified_listing_tag.rb

```diff
@@ -0,0 +1,35 @@
+class ClassifiedListingTag < LiquidTagBase
+  PARTIAL = "classified_listings/liquid".freeze
+
+  def initialize(_tag_name, slug_path_url, _tokens)
+    striped_path = ActionController::Base.helpers.strip_tags(slug_path_url).strip
+    @listing = get_listing(striped_path)
+  end
+
+  def render(_context)
+    ActionController::Base.new.render_to_string(
+      partial: PARTIAL,
+      locals: { listing: @listing },
+    )
+  end
+
+  def get_hash(url)
+    path = Addressable::URI.parse(url).path
+    path.slice!(0, 10) if path.starts_with?("/listings/") # remove leading slash if present
+    path.slice!(-1) if path.ends_with?("/") # remove trailing slash if present
+    Addressable::Template.new("{category}/{slug}").extract(path)&.symbolize_keys
+  end
+
+  def get_listing(url)
+    hash = get_hash(url)
+    raise StandardError, "Invalid URL or slug. Listing not found." if hash.nil?
+
+    listing = ClassifiedListing.find_by(hash)
+    raise StandardError, "Invalid URL or slug. Listing not found." unless listing
+    raise StandardError, "Listing has expired and must be bumped to display as Liquid tag." if listing.bumped_at < (Time.zone.today - 30)
```

Comment:
I understand the meaning of this but there's a corner case I'm thinking about: 

- a person embeds any listing
- the listing expires
- the person edits their own article
- they can't save it anymore because the listing expired

This also means that the article's meaning (let's say the listing is part of the discourse, like an image would be) might be "broken" and removing the expired listing would render the article meaningless.

Wouldn't it be better to instead render a version of the listing tag that says "the listing is expired" instead of raising an error?

Reply:
Yeah I think that makes sense.

Reply:
Good call, @rhymes! That definitely makes sense. I just updated my PR to reflect this implementation.